File: demo2_translate_python_to_go/bank_account.go
package bankaccount

import (
	"fmt"
)

// [AI GENERATED] LLM: GitHub Copilot, Mode: Chat, Date: 2025-12-15
// BankAccount represents a simple checking account with deposit and withdraw operations.
type BankAccount struct {
	name    string
	balance float64
}

// NewBankAccount creates a new BankAccount with the given name and optional initial balance.
func NewBankAccount(name string, balance float64) *BankAccount {
	return &BankAccount{name: name, balance: balance}
}

// Deposit adds the specified amount to the account balance.
// Returns an error if the amount is not positive.
func (a *BankAccount) Deposit(amount float64) error {
	if amount <= 0 {
		return fmt.Errorf("deposit must be positive")
	}
	a.balance += amount
	return nil
}

// Withdraw removes the specified amount from the account balance.
// Returns an error if the amount is not positive or if funds are insufficient.
func (a *BankAccount) Withdraw(amount float64) (float64, error) {
	if amount <= 0 {
		return 0, fmt.Errorf("withdraw must be positive")
	}
	if amount > a.balance {
		return 0, fmt.Errorf("insufficient funds")
	}
	a.balance -= amount
	return amount, nil
}

// String returns a string representation of the account in the format "Name: $Balance".
func (a *BankAccount) String() string {
	return fmt.Sprintf("%s: $%.2f", a.name, a.balance)
}

File: demo2_translate_python_to_go/bank_account.cc
#include "bank_account.h"

// [AI GENERATED] LLM: GitHub Copilot, Mode: Chat, Date: 2025-01-11
BankAccount::BankAccount(const std::string& name, double balance)
    : name_(name), balance_(balance) {}

void BankAccount::Deposit(double amount) {
  if (amount <= 0) {
    throw std::invalid_argument("Deposit must be positive");
  }
  balance_ += amount;
}

double BankAccount::Withdraw(double amount) {
  if (amount <= 0) {
    throw std::invalid_argument("Withdraw must be positive");
  }
  if (amount > balance_) {
    throw std::invalid_argument("Insufficient funds");
  }
  balance_ -= amount;
  return amount;
}

std::string BankAccount::ToString() const {
  std::ostringstream oss;
  oss << name_ << ": $" << std::fixed << std::setprecision(2) << balance_;
  return oss.str();
}
